growthbook/growthbook
# Add Default Assignment Query for Datasources

### Features and Changes

 - Added a `default` tag that will be used to show that datasources and assignment queries as set to default.
 - Add a "Set as default" button in the three-dot menu attached to assignment queries on the parent datasource page.
 - Add an optional `defaultQueryId` on the `settings` object of each datasource. This will be used to mark the default Asssignment Query, if set.

- Closes [#1812](https://github.com/growthbook/growthbook/issues/1812)

### Testing

#### Test 1 (Testing new default UI flag)
1. Set a datasource as default.
2. Open the datasource page. 

 - There should be a teal-blue `default` tag at the top of the page.

#### Test 2 (Testing default assignment query)
1. Set a datasource as default.
2. Open the datasource page. 
3. Scroll to the "Assignment Queries" section.
4. Click the three-dot menu, and click "Set as Default".

 - A teal-blue `default` tag should show next to the Assignment Query name.
 - Create a new Experiment. The new experiment should use the default datasource along with the default assignment query, by default.

### Screenshots
#### Default Datasource Tag
![Default Datasource Tag](https://github.com/growthbook/growthbook/assets/49459061/66103bda-11ba-4ae0-b84c-17fb13c71c3e)
#### Default Assignment Query Tag
![Default Assignment Query Tag](https://github.com/growthbook/growthbook/assets/49459061/287c7890-e291-44b9-b98c-ce661656672a)
